### DISCUSSION TITLE
Can't seem to get forward_mouse_events to work

### DISCUSSION THREAD

**Discussion by rgayler** [bordaigorl/rmview#130] [Q&A] [answered]
Hi. Thanks for the great software.

I installed the latest version of rmview today:

- PC running Ubuntu 21.10
- RM1 v2.11.0.442

Everything seems to be working fine, except that `forward_mouse_events` doesn't seem to work (the reMarkable is connected to the PC via USB). 

I enabled `forward_mouse_events` in `rmview.json`:
```
{
    "ssh": {
        "address": "10.11.99.1",
    	"password": "**********"        
    },
    "orientation": "auto",
    "pen_size": 15,
    "pen_color": "red",
    "pen_trail": 200,
    "forward_mouse_events": true
}
```

The logging shows that it has been picked up from the configuration file:

```
ross@kokopelli:~/RG/projects/rmview-vnc$ rmview -v
2022-01-26 14:45:38+1100 [-] Log opened.
[INFO] STARTING: Wed Jan 26 14:45:38 2022
[INFO] Searching configuration in rmview.json, /home/ross/.config/rmview.json
[DEBUG] Configuration failure in rmview.json: [Errno 2] No such file or directory: 'rmview.json'
[INFO] Fetching configuration from /home/ross/.config/rmview.json
[DEBUG] Config values: {'ssh': {'address': '10.11.99.1', 'password': '**********'}, 'orientation': 'auto', 'pen_size': 15, 'pen_color': 'red', 'pen_trail': 200, 'forward_mouse_events': True}
[INFO] Using known hosts file: /home/ross/.config/rmview_known_hosts
[INFO] Loaded known hosts from /home/ross/.config/rmview_known_hosts
[INFO] Connecting...
[INFO] Connected to 10.11.99.1
[INFO] Detected reMarkable 1.0
[INFO] Using backend 'screenshare'
[INFO] Connecting to ScreenShare, make sure you enable it
[WARNING] Authenticating, please wait...
2022-01-26 14:45:39+1100 [-] ChallengeReaderProtocol starting on 5901
2022-01-26 14:45:39+1100 [-] Starting protocol <rmview.screenstream.screenshare.ChallengeReaderProtocol object at 0x7fefac2955b0>
[DEBUG] Stopping connection worker
[INFO] received timestamp challenge 1643165925605
[INFO] Challenge: 6a1706ad3695692df2bc81e4b332b604e3f4a37f63177a5e0115b74518dc7678, connecting to vnc
2022-01-26 14:45:41+1100 [-] Starting factory <rmview.screenstream.common.VncFactory object at 0x7fefac295cd0>
[DEBUG] Stopping listening for timestamps
[DEBUG] skipping challenge 1643165925605
2022-01-26 14:45:41+1100 [-] (UDP Port 5901 Closed)
2022-01-26 14:45:41+1100 [-] Stopping protocol <rmview.screenstream.screenshare.ChallengeReaderProtocol object at 0x7fefac2955b0>
2022-01-26 14:45:41+1100 [VncClient (TLSMemoryBIOProtocol),client] Using protocol version 3.800
2022-01-26 14:45:41+1100 [VncClient (TLSMemoryBIOProtocol),client] Server: reMarkable rfb
[INFO] Connection to VNC server has been established
```

From the readme on github I presume that with `forward_mouse_events` enabled, mouse clicks and drags on the rmview window on the PC should be sent to the reMarkable tablet as touch events so that you can interact with the rmview window as though it was the tablet.  When I click in the rmview window nothing happens on the tablet. Have I misunderstood what `forward_mouse_events`  is meant to do? If not, is there some diagnostic step I can take to work out what's going wrong?

Thanks.

-------------

As a very minor secondary issue, the logging is somewhat confusing because it appears to claim that `rmview.json` doesn't exist and then proceeds to read from it:

```
[INFO] Searching configuration in rmview.json, /home/ross/.config/rmview.json
[DEBUG] Configuration failure in rmview.json: [Errno 2] No such file or directory: 'rmview.json'
[INFO] Fetching configuration from /home/ross/.config/rmview.json
```

**Comment by bordaigorl** [comment, marked as answer]
Hi, thanks for the feedback.

You are interpreting the meaning of `forward_mouse_events` correctly, the issue is that it was only supported with the custom VNC server (i.e. `"backend": "vncserver"`) which only works on old versions of the tablet's software (below 2.8).
I will update the readme to reflect this.
There were plans to update the custom VNC server to support newer versions but it is very much non trivial to do so.
Since the other backend gives most functionality the plans for updating the other stalled.

---

Regarding the log comment:
- the first line tells you the list of files from which it is going to try to fetch configuration: `rmview.json` (path relative to current directory) and `/home/ross/.config/rmview.json`
- the second line reports that the first path was not found and thus skipped
- the third line reports that the config has been fetched from `/home/ross/.config/rmview.json`
Hope this clarifies.

**Comment by rgayler** [comment]
Thanks @bordaigorl !

> `forward_mouse_events` ... only works on old versions of the tablet's software (below 2.8) ... it is very much non trivial to [update]

Fair enough.

> Regarding the log comment:

Ah - I interpreted `Searching configuration in rmview.json, /home/ross/.config/rmview.json` as meaning that it was searching for file `rmview.json` by looking in `/home/ross/.config/rmview.json`.

Perhaps it would be clearer that it's searching a list of locations by putting a colon after `Searching configuration in`.

Thanks for all your help.
